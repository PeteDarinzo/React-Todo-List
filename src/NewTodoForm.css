
.NewTodoForm {
  padding: 20px;
  display: inline-block;
  border: solid black 2px;
  margin-bottom: 50px;
}

.NewTodoForm-label {
  margin-right: 5px;
}

.NewTodoForm-button {
  padding: 10px;
  margin: 5px 5px 5px 15px;
  background-color: teal;
  border: solid teal 2px;
  color: white;
  border-radius: 10%;
  cursor: pointer;
}

.NewTodoForm-button:hover {
  background-color: white;
  border: solid teal 2px;
  color: teal;
  transition: 250ms;
}

