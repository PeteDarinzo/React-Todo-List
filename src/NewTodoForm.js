import React, { useState } from "react";
import "./NewTodoForm.css";

const NewTodoForm = ({ addTask }) => {
  const initialState = {
    task: "",
    completed: false
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;

    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if (formData.task) {
      addTask({ task: formData.task, completed: false });
      setFormData(initialState);
    }
  }

  return (
    <form className="NewTodoForm" onSubmit={handleSubmit}>
      <label htmlFor="task">New Task</label>
      <input
        id="task"
        type="text"
        name="task"
        value={formData.task}
        onChange={handleChange}
        placeholder="Enter new task"
      />
      <button className="NewTodoForm-button">Submit</button>
    </form>
  );
}

export default NewTodoForm;