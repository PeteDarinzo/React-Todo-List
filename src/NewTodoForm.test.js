import React from "react";
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("renders a new todo form without crashing", () => {
  <NewTodoForm
    addTask={() => null}
  />
});

it("matches the new todo form snapshot", () => {
  const { asFragment } = render(<NewTodoForm
    addTask={() => null}
  />
  );
  expect(asFragment()).toMatchSnapshot();
});
