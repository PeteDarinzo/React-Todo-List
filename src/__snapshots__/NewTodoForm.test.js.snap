// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the new todo form snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="task"
    >
      New Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="Enter new task"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
